src/Reduce.php

```diff
@@ -0,0 +1,178 @@
+     *
+     * Returns null if given collection is empty.
+     *
+     * @param iterable<mixed> $data
+     *
+     * @return mixed|null
+     */
+    public static function toMin(iterable $data)
+    {
+        return static::toValue($data, static function ($carry, $datum) {
+            return min($carry ?? $datum, $datum);
+        });
+    }
+
+    /**
+     * Reduces given iterable to it's max value.
+     *
+     * Items of given collection must be comparable.
+     *
+     * Returns null if given collection is empty.
+     *
+     * @param iterable<mixed> $data
+     *
+     * @return mixed|null
+     */
+    public static function toMax(iterable $data)
+    {
+        return static::toValue($data, static function ($carry, $datum) {
+            return max($carry ?? $datum, $datum);
+        });
+    }
+
+    /**
+     * Reduces given iterable to it's length.
+     *
+     * @param iterable<mixed> $data
+     *
+     * @return int
+     */
+    public static function toCount(iterable $data): int
+    {
+        if (is_array($data) || $data instanceof \Countable) {
+            return count($data);
+        }
+
+        return static::toValue($data, static function ($carry) {
+            return $carry+1;
+        }, 0);
+    }
+
+    /**
+     * Reduces given collection to the sum of it's items.
+     *
+     * @param iterable<numeric> $data
+     *
+     * @return int|float
+     */
+    public static function toSum(iterable $data)
+    {
+        return static::toValue($data, static function ($carry, $datum) {
+            return $carry + $datum;
+        }, 0);
+    }
+
+    /**
+     * Reduces given collection to the product of it's items.
+     *
+     * Returns null if given collection is empty.
+     *
+     * @param iterable<numeric> $data
+     *
+     * @return int|float|null
+     */
+    public static function toProduct(iterable $data)
+    {
+        return static::toValue($data, static function ($carry, $datum) {
+            return ($carry ?? 1) * $datum;
+        });
+    }
+
+    /**
+     * Reduces given collection to the average of it's items.
+     *
+     * Returns null if given collection is empty.
+     *
+     * @param iterable<numeric> $data
+     *
+     * @return int|float|null
+     */
+    public static function toAverage(iterable $data)
+    {
+        [$count, $sum] = static::toValue($data, static function ($carry, $datum): array {
+            [$count, $sum] = $carry;
+            return [$count + 1, $sum + $datum];
+        }, [0, 0]);
+
+        return $count ? ($sum / $count) : null;
+    }
+
+    /**
+     * Returns true if given collection is sorted directly otherwise false.
+     *
+     * Items of given collection must be comparable.
+     *
+     * Returns true if given collection is empty or has one element.
+     *
+     * @param iterable<mixed> $data
+     *
+     * @return bool
+     */
+    public static function isSortedDirectly(iterable $data): bool
```

Comment:
Suggestion: Let's just call this `isSorted`, and the reversed one `isReversed`.

Reply:
Done.